src/ucp/core/ucp_mm.c

```diff
@@ -424,34 +393,32 @@ ucp_memh_get_common(ucp_context_h context, void *address, size_t length,
         ucs_trace("registered address %p length %zu on md[%d]=%s %p",
                   reg_address, reg_length, md_index,
                   context->tl_mds[md_index].rsc.md_name,
-                  region->memh.uct[md_index]);
-        new_md_map_ok |= UCS_BIT(md_index);
+                  memh->uct[md_index]);
+        md_map_registered |= UCS_BIT(md_index);
     }
 
-    region->memh.md_map |= new_md_map_ok;
-    *memh_p = &region->memh;
+    memh->md_map |= md_map_registered;
+    *memh_p       = memh;
     return UCS_OK;
 }
 
-static void ucp_memh_put_common(ucp_context_h context, ucp_mem_h memh, int inv)
+static void ucp_memh_put_common(ucp_context_h context, ucp_mem_h memh,
```

Comment:
can we make most of ucp_memh_put/common inline as well, at least for rcache!=NULL case? it's also a datapath